### PR TITLE
Anshu-Assignment-17-40min-Done

### PR DESCRIPTION
**Pull request by AnshuKJun20**
image
image

### CODE REVIEW

File: src/anshuKJun20/CharacterIndex.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	
+ void findIndex(String name) {
+	 for(int index = 0 ;index < name.length();index++) {
+		 int count=0;
+		 char ch = name.charAt(index);
+		 if(name.indexOf(ch)== index) {
+		    for(int i =0;i<name.length();i++) {
+		        if(name.charAt(i)== ch) {
+			      count++;
+			        if(count>1) {
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/CharacterIndex.java]: You have increased the complexity in code, Using one for loop and one if block you can achieve the output.

File: src/anshuKJun20/CountMultipleCharacter.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	   int countdigit =0,countSpecialChar=0, countUpperCase=0,countLowerCase =0,countletters =0;
+	   for(int index = 0; index<ch.length;index++) {
+		   if(Character.isDigit(ch[index])==true) {
+			   countdigit++;
+		   } else if(Character.isLetter(ch[index])){
+			   countletters++;
+			   if(Character.isLowerCase(ch[index])==true) {
+				   countLowerCase++;
+				} if(Character.isUpperCase(ch[index])==true) {
+					countUpperCase++;
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/CountMultipleCharacter.java:13]: This else if block is wrong. Please think differently here. countletters without using this varibale you can achive expected output.

File: src/anshuKJun20/CountMultipleCharacter.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			   countSpecialChar++;
+		   }
+	   }
+	   
+	   System.out.println("Total Digit is:"+countdigit+"'\n' Total Letters:"+countletters+"'\n' Total Upper Case:"+countUpperCase+"'\n' Total Lower Case:"+countLowerCase+"'\n' Total Special character:"+countSpecialChar);
+   }
+   public static void main(String[] args) {
+	   CountMultipleCharacter countmultiplecharcter = new CountMultipleCharacter();
+	   char[] ch = {'1','2','k','r','#','E','f','(','F','3','&'};
+	      countmultiplecharcter.findcharcter(ch);
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/CountMultipleCharacter.java]: This is also wrong, Here you don't have to convert in char, You have to pass string here.

File: src/anshuKJun20/CountMultipleCharacter.java (hunk 2 above)
Comment (AnshuKJun20) [reply src/anshuKJun20/CountMultipleCharacter.java:13]: Done
old

File: src/anshuKJun20/CharacterIndex.java (hunk 1 above)
Comment (AnshuKJun20) [reply src/anshuKJun20/CharacterIndex.java]: Done 
oldassg

File: src/anshuKJun20/CountMultipleCharacter.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+			   countdigit++;
+		       }else if(Character.isLowerCase(ch)==true) {
+				   countLowerCase++;
+		       }else if(Character.isUpperCase(ch)==true) {
+					countUpperCase++;
+				}else {
+			   countSpecialChar++;
+			   }
+	         }
+	     System.out.println("Total Digit is:"+countdigit+"'\n' Total Upper Case:"+countUpperCase+"'\n' Total Lower Case:"+countLowerCase+"'\n' Total Special character:"+countSpecialChar);
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/CountMultipleCharacter.java:18]: Missed the printing statement of letter.

**Pull request merged**